packages/eslint-plugin-svelte/src/rules/no-navigation-without-base.ts

```diff
@@ -0,0 +1,331 @@
+			}
+		],
+		messages: {
+			gotoNotPrefixed: "Found a goto() call with a url that isn't prefixed with the base path.",
+			linkNotPrefixed: "Found a link with a url that isn't prefixed with the base path.",
+			pushStateNotPrefixed:
+				"Found a pushState() call with a url that isn't prefixed with the base path.",
+			replaceStateNotPrefixed:
+				"Found a replaceState() call with a url that isn't prefixed with the base path."
+		},
+		type: 'suggestion'
+	},
+	create(context) {
+		let basePathNames: Set<TSESTree.Identifier> = new Set<TSESTree.Identifier>();
+		return {
+			Program() {
+				const referenceTracker = new ReferenceTracker(
+					getSourceCode(context).scopeManager.globalScope!
+				);
+				basePathNames = extractBasePathReferences(referenceTracker, context);
+				const {
+					goto: gotoCalls,
+					pushState: pushStateCalls,
+					replaceState: replaceStateCalls
+				} = extractFunctionCallReferences(referenceTracker);
+				if (context.options[0]?.ignoreGoto !== true) {
+					for (const gotoCall of gotoCalls) {
+						checkGotoCall(context, gotoCall, basePathNames);
+					}
+				}
+				if (context.options[0]?.ignorePushState !== true) {
+					for (const pushStateCall of pushStateCalls) {
+						checkShallowNavigationCall(
+							context,
+							pushStateCall,
+							basePathNames,
+							'pushStateNotPrefixed'
+						);
+					}
+				}
+				if (context.options[0]?.ignoreReplaceState !== true) {
+					for (const replaceStateCall of replaceStateCalls) {
+						checkShallowNavigationCall(
+							context,
+							replaceStateCall,
+							basePathNames,
+							'replaceStateNotPrefixed'
+						);
+					}
+				}
+			},
+			SvelteAttribute(node) {
+				if (
+					context.options[0]?.ignoreLinks === true ||
+					node.parent.parent.type !== 'SvelteElement' ||
+					node.parent.parent.kind !== 'html' ||
+					node.parent.parent.name.type !== 'SvelteName' ||
+					node.parent.parent.name.name !== 'a' ||
+					node.key.name !== 'href'
+				) {
+					return;
+				}
+				const hrefValue = node.value[0];
+				if (hrefValue.type === 'SvelteLiteral') {
+					if (!expressionIsAbsolute(hrefValue)) {
+						context.report({ loc: hrefValue.loc, messageId: 'linkNotPrefixed' });
+					}
+					return;
+				}
+				if (
+					!expressionStartsWithBase(context, hrefValue.expression, basePathNames) &&
+					!expressionIsAbsolute(hrefValue.expression)
+				) {
+					context.report({ loc: hrefValue.loc, messageId: 'linkNotPrefixed' });
+				}
+			}
+		};
+	}
+});
+
+// Extract all imports of the base path
+
+function extractBasePathReferences(
+	referenceTracker: ReferenceTracker,
+	context: RuleContext
+): Set<TSESTree.Identifier> {
+	const set = new Set<TSESTree.Identifier>();
+	for (const { node } of referenceTracker.iterateEsmReferences({
+		'$app/paths': {
+			[ReferenceTracker.ESM]: true,
+			base: {
+				[ReferenceTracker.READ]: true
+			}
+		}
+	})) {
+		const variable = findVariable(context, (node as TSESTree.ImportSpecifier).local);
```

Comment:
Could you please add a test case to see if the following import works?

```js
	import *  as paths from '$app/paths';
    paths.base;
```

Reply:
Great tip, found an issue with the implementation as well